packages/cli/src/lib/util/rooms.ts

```diff
@@ -0,0 +1,51 @@
+import { CLIError } from '@oclif/errors'
+import { APICommand, selectFromList } from '@smartthings/cli-lib'
+import { LocationItem, Room, SmartThingsClient } from '@smartthings/core-sdk'
+import * as roomsUtil from './rooms'
+
+
+export const tableFieldDefinitions = ['name', 'locationId', 'roomId']
+
+export async function getRoomsByLocation(client: SmartThingsClient, locationId?: string): Promise<RoomWithLocation[]> {
+	let locations: LocationItem[] = []
+	if (locationId) {
+		locations = [await client.locations.get(locationId)]
+	} else {
+		locations = await client.locations.list()
+	}
+
+	if (!locations || locations.length == 0) {
+		throw new CLIError('could not find any locations for your account. Perhaps ' +
+			"you haven't created any locations yet.")
+	}
+
+	let rooms: RoomWithLocation[] = []
+	for (const location of locations) {
+		const locationRooms = await client.rooms.list(location.locationId)
+		rooms = rooms.concat(locationRooms.map(room => { return { ...room, locationName: location.name } }))
+	}
+	return rooms
+}
+
+export type RoomWithLocation = Room & {
+	locationName?: string
+}
+
+export async function chooseRoom(command: APICommand, locationId?: string, roomArg?: string): Promise<[string, string]> {
+	const rooms = await roomsUtil.getRoomsByLocation(command.client, locationId)
```

Comment:
This line and the circular import are the only code changes to this moved file. Calling `getRoomsByLocation` from an import reference enabled unit testing the module easier without needing to split up the related utility functions.